apps/anomalycor/server/dataFunctions/data_contour_diff.js

```diff
@@ -92,100 +109,111 @@ dataContourDiff = function (plotParams, plotFunction) {
+    curve.unitKey = axisKey;
 
-    var d;
-    // this is a database driven curve, not a difference curve
-    // prepare the query from the above parameters
-    let statement =
-      "{{xValClause}} " +
-      "{{yValClause}} " +
-      "count(distinct {{dateString}}) as N_times, " +
-      "min({{dateString}}) as min_secs, " +
-      "max({{dateString}}) as max_secs, " +
-      "{{statisticClause}} " +
-      "{{queryTableClause}} " +
-      "where 1=1 " +
-      "{{dateClause}} " +
-      "{{variableClause}} " +
-      "{{validTimeClause}} " +
-      "{{forecastLengthClause}} " +
-      "{{levelClause}} " +
-      "group by xVal,yVal " +
-      "order by xVal,yVal" +
-      ";";
+    let d;
+    if (!diffFrom) {
+      let queryResult;
+      const startMoment = moment();
+      let finishMoment;
+      try {
+        statement =
+          "{{xValClause}} " +
+          "{{yValClause}} " +
+          "count(distinct {{dateString}}) as N_times, " +
+          "min({{dateString}}) as min_secs, " +
+          "max({{dateString}}) as max_secs, " +
+          "{{statisticClause}} " +
+          "{{queryTableClause}} " +
+          "where 1=1 " +
+          "{{dateClause}} " +
+          "{{variableClause}} " +
+          "{{validTimeClause}} " +
+          "{{forecastLengthClause}} " +
+          "{{levelClause}} " +
+          "group by xVal,yVal " +
+          "order by xVal,yVal" +
+          ";";
 
-    statement = statement.replace("{{xValClause}}", xValClause);
-    statement = statement.replace("{{yValClause}}", yValClause);
-    statement = statement.replace("{{statisticClause}}", statisticClause);
-    statement = statement.replace("{{queryTableClause}}", queryTableClause);
-    statement = statement.replace("{{variableClause}}", variableClause);
-    statement = statement.replace("{{validTimeClause}}", validTimeClause);
-    statement = statement.replace("{{forecastLengthClause}}", forecastLengthClause);
-    statement = statement.replace("{{levelClause}}", levelClause);
-    statement = statement.replace("{{dateClause}}", dateClause);
-    statement = statement.split("{{dateString}}").join(dateString);
-    dataRequests[label] = statement;
+        statement = statement.replace("{{xValClause}}", xValClause);
+        statement = statement.replace("{{yValClause}}", yValClause);
+        statement = statement.replace("{{statisticClause}}", statisticClause);
+        statement = statement.replace("{{queryTableClause}}", queryTableClause);
+        statement = statement.replace("{{variableClause}}", variableClause);
+        statement = statement.replace("{{validTimeClause}}", validTimeClause);
+        statement = statement.replace("{{forecastLengthClause}}", forecastLengthClause);
+        statement = statement.replace("{{levelClause}}", levelClause);
+        statement = statement.replace("{{dateClause}}", dateClause);
+        statement = statement.split("{{dateString}}").join(dateString);
+        dataRequests[label] = statement;
```

Comment:
Is there a reason we needed to do `statement.replace()` here and we can't use a [string literal?](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) They should behave similarly to Python's f-strings.

E.g.
```js
statement = `
    ${xValClause}  +
    ${yValClause}  +
    count(distinct ${dateString}) as N_times,  +
    min(${dateString}) as min_secs,  +
    max(${dateString}) as max_secs,  +
    ${statisticClause}  +
    ${queryTableClause}  +
    where 1=1  +
    ${dateClause}  +
    ${variableClause}  +
    ${validTimeClause}  +
    ${forecastLengthClause}  +
    ${levelClause}  +
    group by xVal,yVal  +
    order by xVal,yVal +
    ;`;
```

I think the big downside with template literals is the literal is evaluated when the string is defined. That doesn't seem like it'd be an issue here. They also preserve newlines.

Reply:
This is the way that Randy (and later Gopa in the couchbase app) wanted to do things. I can create an issue to change these in the future for the mysql apps (will take way too long to do before the cb-metar app needs to be done), but we're committed to this in the couchbase app because Gopa wanted to have templates stored in files, and I don't think string literals are interpreted correctly when reading from a template.

Reply:
I'd be alright with opening another issue to address this. It sounds like I'm missing some context here. Your comment makes me wonder if we should just use a templating engine for our N1QL & SQL queries. 

The multi `.replace()` approach definitely works but there must be a way to make the `statement` more concise. I could see it being relatively easy to forget a `.replace()` statement.

Reply:
I've added this to my Finish Linting the Apps issue. Will work on it when the cb-metar app is done.